core/src/test/java/org/apache/druid/timeline/DataSegmentTest.java

```diff
@@ -145,7 +146,7 @@ public void testV1Serialization() throws Exception
     Assert.assertEquals(ImmutableMap.of("type", "numbered", "partitionNum", 3, "partitions", 0), objectMap.get("shardSpec"));
     Assert.assertEquals(TEST_VERSION, objectMap.get("binaryVersion"));
     Assert.assertEquals(1, objectMap.get("size"));
-    Assert.assertEquals(4, ((Map) objectMap.get("lastCompactionState")).size());
+    Assert.assertEquals(5, ((Map) objectMap.get("lastCompactionState")).size());
```

Comment:
this is a test for "V1" Shouldn't this test remain without the `filter` being part of the CompactionState?

Reply:
Scratch that. It's v1 of DataSegment. This seems fine.